src/arithmetic/boolean_funcs/boolean_funcs.c

```diff
@@ -115,11 +111,9 @@ SIValue AR_LT(SIValue *argv, int argc, void *private_data) {
 	if(disjointOrNull == COMPARED_NULL) {
 		// Comparisons with NULL values always return NULL.
 		return SI_NullVal();
-	} else if(disjointOrNull == DISJOINT) {
-		// Emit error when attempting to compare invalid types
-		Error_SITypeMismatch(b, SI_TYPE(a));
-		return SI_NullVal(); // The return doesn't matter, as the caller will check for errors.
 	}
+	if(disjointOrNull == COMPARED_NAN) return SI_BoolVal(false);
+	if(disjointOrNull == DISJOINT) return SI_NullVal();
```

Comment:
due to the reparative nature of this code, consider adding a macro:
```c
#define HANDLE_DISJOINT_CMP(disjointOrNull) \
 	if((disjointOrNull) == COMPARED_NULL || (disjointOrNull) == COMPARED_NAN) return SI_BoolVal(false); \
 	if((disjointOrNull) == DISJOINT) return SI_NullVal();
```

Reply:
it will be hard to debug